mod/wikirate/set/right/citation_count.rb

```diff
@@ -20,10 +20,17 @@ def default_related_overview_modal_box_args args
         text: link_text,
         html_args: { class: "#{'no-citations' if no_citations?(args)}" }
       )
-    return unless parent && (pp = parent.parent) && (ppl = pp.card.left) &&
-                  ppl.type_id == WikirateCompanyID
+    return unless (company = on_company_page?)
     link_args[:path_opts] ||= {}
-    link_args[:path_opts][:company] = ppl.key
+    link_args[:path_opts][:company] = company.key
+    link_args[:path_opts][:general_overview] = true
+  end
+
+  def on_company_page?
+    return unless parent && (pp = parent.parent) && (ppc = pp.card)
+    return ppc if ppc.type_id == WikirateCompanyID
+    return unless (ppl = ppc.left) && ppl.type_id == WikirateCompanyID
+    return ppl
```

Comment:
suggest case...when  statement.

...and we need better variable names.

more importantly, this code looks really fragile to me.